tests/components/homekit/test_diagnostics.py

```diff
@@ -290,3 +292,321 @@ async def test_config_entry_accessory(
+    assert await async_setup_component(hass, "demo", {"demo": {}})
+    hk_driver.publish = MagicMock()
+
+    demo_config_entry = MockConfigEntry(domain="domain")
+    demo_config_entry.add_to_hass(hass)
+    assert await async_setup_component(hass, "demo", {"demo": {}})
+    await hass.async_block_till_done()
+
+    entry = entity_reg.async_get("light.ceiling_lights")
+    assert entry is not None
+    device_id = entry.device_id
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_NAME: "mock_name",
+            CONF_PORT: 12345,
+            CONF_DEVICES: [device_id],
+            "filter": {
+                "exclude_domains": [],
+                "exclude_entities": [],
+                "include_domains": [],
+                "include_entities": ["light.none"],
+            },
+        },
+    )
+    entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(entry.entry_id)
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_STARTED)
+    await hass.async_block_till_done()
+    diag = await get_diagnostics_for_config_entry(hass, hass_client, entry)
+    diag.pop("iid_storage")
+    diag.pop("bridge")
+    assert diag == {
+        "accessories": [
+            {
+                "aid": 1,
+                "services": [
+                    {
+                        "characteristics": [
+                            {"format": "bool", "iid": 2, "perms": ["pw"], "type": "14"},
+                            {
+                                "format": "string",
+                                "iid": 3,
+                                "perms": ["pr"],
+                                "type": "20",
+                                "value": "Home Assistant",
+                            },
+                            {
+                                "format": "string",
+                                "iid": 4,
+                                "perms": ["pr"],
+                                "type": "21",
+                                "value": "Bridge",
+                            },
+                            {
+                                "format": "string",
+                                "iid": 5,
+                                "perms": ["pr"],
+                                "type": "23",
+                                "value": "mock_name",
+                            },
+                            {
+                                "format": "string",
+                                "iid": 6,
+                                "perms": ["pr"],
+                                "type": "30",
+                                "value": "homekit.bridge",
+                            },
+                            {
+                                "format": "string",
+                                "iid": 7,
+                                "perms": ["pr"],
+                                "type": "52",
+                                "value": "2022.12.0",
```

Comment:
🤦 
https://github.com/home-assistant/core/pull/82201

Reply:
hehe lol, stuff happens. Will pull it in. Thanks for being on top of it!